File: j2ee-spring-patterns/mvc/python/BookController.py
class BookController:
    def __init__(self, book, view):
        self.book = book
        self.view = view

    def update_view(self, book):
        self.view.print_book(book)

    def set_name(self, book_name):
        self.book.name = book_name

    def set_author(self, book_author):
        self.book.author = book_author

File: j2ee-spring-patterns/mvc/go/controller/book_controller.go
package controller

import (
	"mvc/model"
	"mvc/view"
)

type BookController struct {
	Book model.Book
	View view.BookView
}

func (controller *BookController) SetTitle(name string) {
	controller.Book.Title = name
}

func (controller *BookController) SetAuthor(author string) {
	controller.Book.Author = author
}

func (controller *BookController) UpdateView() {
	controller.View.PrintBook(controller.Book)
}
